### DISCUSSION TITLE
Automerge after approving review

### DISCUSSION THREAD

**Discussion by elijaholmos** [renovatebot/renovate#17796] [Request Help] [answered]
I've been unable to find any other resources that seem to answer the specific question I have about renovate's functionality:

Can I configure renovate to automerge a GitHub PR after it has been approved and all tests have passed?

I've tried combining GitHub branch protection & renovate config with the following options:
![image](https://user-images.githubusercontent.com/35435704/190223195-a41f0b36-92e6-4481-8eea-d0cea8cf1697.png)
```json
{
	"extends": [
		":ignoreModulesAndTests",
		":autodetectPinVersions",
		":prConcurrentLimit10",
		"group:recommended",
		"workarounds:all",
		":semanticCommitType(chore)"
	],
	"automerge": true,
	"automergeStrategy": "squash"
}
```

However, after approving a PR, renovate does not auto-merge, even after checks have successfully completed. Is the functionality I'm trying to achieve possible? If so, what values should I alter to achieve it?

**Comment by rarkins** [comment, marked as answer]
Have you looked into the `platformAutomerge` feature? This way Renovate will enable GitHub's native automerge, which should behave as you describe.

**Reply by elijaholmos** [reply]
That looks pretty promising, thank you for mentioning it I'll give it a try shortly. I do see a note about some incompatibility with `rebaseWhen=auto`, regarding the rebase behavior not being guaranteed. If I manually set `rebaseWhen=behind-base-branch`, will the rebasing be guaranteed before automerge?

**Reply by viceice** [reply]
no, you need to use a branch protection to force an uptodate branch

**Reply by elijaholmos** [reply]
Seems to function fine - there's just a few hour delay from when the PR is approved to when renovate performs the merge, which I thought were false negatives during my testing. Thank you gentlemen!
